Content.Shared/Nutrition/Components/HungerComponent.cs

```diff
@@ -0,0 +1,162 @@
+    /// </summary>
+    [DataField("thresholds", customTypeSerializer: typeof(DictionarySerializer<HungerThreshold, float>))]
+    public Dictionary<HungerThreshold, float> Thresholds = new()
+    {
+        { HungerThreshold.Overfed, 200.0f },
+        { HungerThreshold.Okay, 150.0f },
+        { HungerThreshold.Peckish, 100.0f },
+        { HungerThreshold.Starving, 50.0f },
+        { HungerThreshold.Dead, 0.0f }
+    };
+
+    /// <summary>
+    /// A dictionary relating hunger thresholds to corresponding alerts.
+    /// </summary>
+    [DataField("hungerThresholdAlerts", customTypeSerializer: typeof(DictionarySerializer<HungerThreshold, AlertType>))]
+    public Dictionary<HungerThreshold, AlertType> HungerThresholdAlerts = new()
+    {
+        { HungerThreshold.Peckish, AlertType.Peckish },
+        { HungerThreshold.Starving, AlertType.Starving },
+        { HungerThreshold.Dead, AlertType.Starving }
+    };
+
+    /// <summary>
+    /// A dictionary relating HungerThreshold to how much they modify <see cref="BaseDecayRate"/>.
+    /// </summary>
+    [DataField("hungerThresholdDecayModifiers", customTypeSerializer: typeof(DictionarySerializer<HungerThreshold, float>))]
+    public Dictionary<HungerThreshold, float> HungerThresholdDecayModifiers = new()
+    {
+        { HungerThreshold.Overfed, 1.2f },
+        { HungerThreshold.Okay, 1f },
+        { HungerThreshold.Peckish, 0.8f },
+        { HungerThreshold.Starving, 0.6f },
+        { HungerThreshold.Dead, 0.6f }
+    };
+
+    /// <summary>
+    /// The amount of slowdown applied when an entity is starving
+    /// </summary>
+    [DataField("starvingSlowdownModifier"), ViewVariables(VVAccess.ReadWrite)]
+    public float StarvingSlowdownModifier = 0.75f;
+
+    /// <summary>
+    /// Damage dealt when your current threshold is at HungerThreshold.Dead
+    /// </summary>
+    [DataField("starvationDamage")]
+    public DamageSpecifier? StarvationDamage;
+
+    /// <summary>
+    /// The time when the hunger will update next.
+    /// </summary>
+    [DataField("nextUpdateTime"), ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan NextUpdateTime;
+
+    /// <summary>
+    /// The time between each update.
+    /// </summary>
+    [DataField("updateRate"), ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan UpdateRate = TimeSpan.FromSeconds(1);
+}
+
+[Serializable, NetSerializable]
+public sealed class HungerComponentState : ComponentState
+{
+    public float CurrentHunger;
+
+    public float BaseDecayRate;
+
+    public float ActualDecayRate;
+
+    public HungerThreshold LastHungerThreshold;
+
+    public HungerThreshold CurrentThreshold;
+
+    public Dictionary<HungerThreshold, float> HungerThresholds;
+
+    public Dictionary<HungerThreshold, AlertType> HungerAlertThresholds;
+
+    public float StarvingSlowdownModifier;
+
+    public TimeSpan NextUpdateTime;
+
+    public TimeSpan UpdateRate;
+
+    public HungerComponentState(float currentHunger,
```

Comment:
This is a lot of data for something every tick? If you want to keep all of this around at least just split currenthunger off to its own component or derive it.